src/psyclone/gocean1p0.py

```diff
@@ -404,86 +411,146 @@ def __init__(self, parent=None,
+                GOLoop._bounds_lookup['offset_any'][gridpt_type][itspace] = \
                     {'inner': {'start': "1", 'stop': "{stop}"},
                      'outer': {'start': "1", 'stop': "{stop}"}}
         # For 'every' grid-point type
         for offset in SUPPORTED_OFFSETS:
             for itspace in VALID_ITERATES_OVER:
-                self._bounds_lookup[offset]['every'][itspace] = \
+                GOLoop._bounds_lookup[offset]['every'][itspace] = \
                     {'inner': {'start': "1", 'stop': "{stop}+1"},
                      'outer': {'start': "1", 'stop': "{stop}+1"}}
 
+    # -------------------------------------------------------------------------
+    @staticmethod
+    def add_bounds(bound_info):
+        '''
+        Adds a new iteration space to PSyclone. An iteration space in the
+        gocean1.0 API is for a certain offset type and field type. It defines
+        the loop boundaries for the outer and inner loop. The format is a
+        ":" separated tuple:
+        bound_info = offset-type:field-type:iteration-space:outer-start:
+                      outer-stop:inner-start:inner-stop
+        Example:
+        bound_info = offset_ne:ct:all_pts:outer:1:{stop}+1:2:{stop}
+
+        :param str bound_info: A string that contains a ":" separated \
+               tuple with the iteration space definition.
+        :raises ValueError: if bound_info is not a string.
+        :raises ConfigurationError: if bound_info is not formatted correctly.
+        '''
+
+        if not isinstance(bound_info, str):
+            raise ValueError("The parameter 'bound_info' must be a string, "
+                             "got '{0}' (type {1}".format(bound_info,
+                                                          type(bound_info)))
+
+        data = bound_info.split(":")
+        if len(data) != 7:
+            from psyclone.configuration import ConfigurationError
+            raise ConfigurationError("An iteration space must be in the form "
+                                     "\"offset-type:field-type:"
+                                     "iteration-space:outer-start:"
+                                     "outer-stop:inner-start:inner-stop\"\n"
+                                     "It was \"{0}\"".format(bound_info))
+
+        if len(GOLoop._bounds_lookup) == 0:
+            GOLoop.setup_bounds()
+
+        current_bounds = GOLoop._bounds_lookup   # Shortcut
+        # Check offset-type exists
+        if not data[0] in current_bounds:
+            current_bounds[data[0]] = {}
+
+        # Check field-type exists
+        if not data[1] in current_bounds[data[0]]:
+            current_bounds[data[0]][data[1]] = {}
+
+        # Check iteration space exists:
+        if not data[2] in current_bounds[data[0]][data[1]]:
+            current_bounds[data[0]][data[1]][data[2]] = {}
+            VALID_ITERATES_OVER.append(data[2])
+        current_bounds[data[0]][data[1]][data[2]] = \
```

Comment:
I can't really test the offset and field types, since I want to be able to support new field types etc (i.e. different things than just dl_esm_inf), so I left this open on purpose.
For start and stop - that's difficult. I've added a test that all "{...} expressions are "{start}" and "{stop"} only, but otherwise I am not sure what to check - maybe a different infrastructure library will have a value for halo size, so the start index could be something like: "{start}-halo_size" (halo size being somehow set depending on the field). I could try to parse the expression, but I am not sure if this is really worth the effort?